lib/GHCup/GHC.hs

```diff
@@ -1037,6 +1042,24 @@ compileGHC targetGhc crossTarget vps bstrap hghc jobs mbuildConfig patches aargs
     pure installVer
 
  where
+  bootAndConfigure :: ( MonadReader env m
+                      , HasSettings env
+                      , HasDirs env
+                      , HasLog env
+                      , MonadIO m
+                      , MonadThrow m
+                      )
+                   => GHCupPath
+                   -> Excepts '[ProcessError, NotFoundInPATH] m ()
+  bootAndConfigure tmpUnpack = do
+    let bootFile = fromGHCupPath tmpUnpack </> "boot"
+    hasBootFile <- liftIO $ doesFileExist bootFile
+    when hasBootFile $ do
+      lift $ logDebug "Doing ghc-bootstrap"
+      python3 <- liftE $ makeAbsolute "python3"
```

Comment:
This is not necessary. The reason we swallow the bootstrap error is because of:

```hs
          tver <- liftE $ catchAllE @_ @'[ProcessError, ParseError, NotFoundInPATH] @'[] (\_ -> pure Nothing) $ fmap Just $ getGHCVer
```

`execLogged` already returns the correct `Left` when `executeFile` can't find the exe. So separating bootstrap from retrieving the GHC version should be enough.

Reply:
reverted this